src/fragments/lib/datastore/native_common/advanced-workflows.mdx

```diff
@@ -0,0 +1,272 @@
+
+<Fragments fragments={{android: android1}} />
+
+import flutter1 from "/src/fragments/lib/datastore/flutter/advanced-workflows/custom-primary-key/delete-snippet.mdx";
+
+<Fragments fragments={{flutter: flutter1}} />
+
+import ios1 from "/src/fragments/lib/datastore/ios/advanced-workflows/custom-primary-key/delete-snippet.mdx";
+
+<Fragments fragments={{ios: ios1}} />
+
+import js1 from "/src/fragments/lib/datastore/js/advanced-workflows/custom-primary-key/delete-snippet.mdx";
+
+<Fragments fragments={{js: js1}} />
+
+### Many-to-many
+
+A schema using custom primary keys with a `many-to-many` relationship is converted into a `hasMany` / `belongsTo` relationship with a join table.
+
+For instance, with the following schema:
+```graphql
+type Post @model {
+	customPostId: ID! @primaryKey(sortKeyFields: ["title"])
+	title: String!
+	content: String
+	tags: [Tag] @manyToMany(relationName: "PostTags")
+}
+
+type Tag @model {
+	customTagId: ID! @primaryKey(sortKeyFields: ["label"])
+	label: String!
+	posts: [Post] @manyToMany(relationName: "PostTags")
+}
+```
+
+Codegen will generate the following models (NOTE: if we keep this section, we will split this up and create snippets for each platform):
+```graphql
+export declare class Post {
+  readonly [__modelMeta__]: {
+    identifier: ManagedIdentifier<Post, 'id'>;
+    readOnlyFields: 'createdAt' | 'updatedAt';
+  };
+  readonly id: string;
+  readonly customPostId: string;
+  readonly title: string;
+  readonly content?: string | null;
+  readonly tags?: (PostTags | null)[] | null;
+  readonly createdAt?: string | null;
+  readonly updatedAt?: string | null;
+  constructor(init: ModelInit<Post>);
+  static copyOf(source: Post, mutator: (draft: MutableModel<Post>) => MutableModel<Post> | void): Post;
+}
+
+export declare class Tag {
+  readonly [__modelMeta__]: {
+    identifier: CompositeIdentifier<Tag, ['customTagId', 'label']>;
+    readOnlyFields: 'createdAt' | 'updatedAt';
+  };
+  readonly customTagId: string;
+  readonly label: string;
+  readonly posts?: (PostTags | null)[] | null;
+  readonly createdAt?: string | null;
+  readonly updatedAt?: string | null;
+  constructor(init: ModelInit<Tag>);
+  static copyOf(source: Tag, mutator: (draft: MutableModel<Tag>) => MutableModel<Tag> | void): Tag;
+}
+
+export declare class PostTags {
+  readonly [__modelMeta__]: {
+    identifier: ManagedIdentifier<PostTags, 'id'>;
+    readOnlyFields: 'createdAt' | 'updatedAt';
+  };
+  readonly id: string;
+  readonly postID: string;
+  readonly tagID: string;
+  readonly taglabel: string;
+  readonly post: Post;
+  readonly tag: Tag;
+  readonly createdAt?: string | null;
+  readonly updatedAt?: string | null;
+  constructor(init: ModelInit<PostTags>);
+  static copyOf(source: PostTags, mutator: (draft: MutableModel<PostTags>) => MutableModel<PostTags> | void): PostTags;
+}
+```
+
+### Examples
+TODO: link to complete sample apps using CPK?
```

Comment:
Thoughts?

Reply:
Wouldn't the apps need to be professional quality if we do so?